src/mono/wasm/Wasm.Build.Tests/HybridGlobalizationTests.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class HybridGlobalizationTests : BuildTestBase
+    {
+        public HybridGlobalizationTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+            : base(output, buildContext)
+        {
+        }
+
+        public static IEnumerable<object?[]> HybridGlobalizationTestData(bool aot, RunHost host)
+            => ConfigWithAOTData(aot)
+                .WithRunHosts(host)
+                .UnwrapItemsAsArrays();
+
+        [Theory]
+        [MemberData(nameof(HybridGlobalizationTestData), parameters: new object[] { /*aot*/ false, RunHost.All })]
+        [MemberData(nameof(HybridGlobalizationTestData), parameters: new object[] { /*aot*/ true, RunHost.All })]
+        public void AOT_HybridGlobalizationTests(BuildArgs buildArgs, RunHost host, string id)
+            => TestHybridGlobalizationTests(buildArgs, host, id);
+
+        [Theory]
+        [MemberData(nameof(HybridGlobalizationTestData), parameters: new object[] { /*aot*/ false, RunHost.All })]
+        public void RelinkingWithoutAOT(BuildArgs buildArgs, RunHost host, string id)
+            => TestHybridGlobalizationTests(buildArgs, host, id,
+                                            extraProperties: "<WasmBuildNative>true</WasmBuildNative>",
+                                            dotnetWasmFromRuntimePack: false);
+
+        private void TestHybridGlobalizationTests(BuildArgs buildArgs, RunHost host, string id, string extraProperties="", bool? dotnetWasmFromRuntimePack=null)
+        {
+            string projectName = $"hybrid";
+            extraProperties = $"{extraProperties}<HybridGlobalization>true</HybridGlobalization>";
+
+            buildArgs = buildArgs with { ProjectName = projectName };
+            buildArgs = ExpandBuildArgs(buildArgs, extraProperties);
+
+            if (dotnetWasmFromRuntimePack == null)
+                dotnetWasmFromRuntimePack = !(buildArgs.AOT || buildArgs.Config == "Release");
+
+            string programText = @"
```

Comment:
Please, use C# raw literals (`"""`) or even better, can we have it as a .cs file in testassets?

Reply:
Do you mean to change the convention for all WBT? Other tests have the test code embedded in the test file, it would be messy to move only this one to assets.

Reply:
Some of the WBT has assets there, like `Blazor_net50`. WBT has `BuildEnvironment.TestAssetsPath` for accessing items from there. In general, we should move from having these pieces of code in literals and instead have a set of prepared test apps.